analyzers/src/SonarAnalyzer.CSharp/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -43,6 +44,25 @@ internal static class IMethodSymbolExtensions
             && invokedMethod.ContainingType != null
             && IsObjectOrType(invokedMethod.ContainingType);
 
+        internal static bool HasName(this IMethodSymbol methodSymbol, string name)
```

Comment:
Why not to use `IsKnownType` around here?

Reply:
`IsKnownType` needs the `SemanticModel` and it would be inefficient to compute the symbols again.

Reply:
This is not going to work for everything because of nested classes, local functions, generic stuff and other weird stuff. We should reuse what we already have. 
While looking inside `SymbolHelper.IsKnownType`, there already is `symbolType.Is(knownType)` extension that should solve the main problem. This should be invoked somehow like `methodSymbol.Is(KnownType_System_Threading_Monitor, "Enter")`